newIDE/app/src/ProjectsStorage/LocalFileStorageProvider/LocalProjectWriter.js

```diff
@@ -180,16 +185,49 @@ export const onSaveProject = async (
   };
 };
 
-export const onChooseSaveProjectAsLocation = async ({
+export const generateOnChooseSaveProjectAsLocation = ({
+  setDialog,
+  closeDialog,
+}: {
+  setDialog: (() => React.Node) => void,
+  closeDialog: () => void,
+}) => async ({
   project,
   fileMetadata,
 }: {|
   project: gdProject,
   fileMetadata: ?FileMetadata, // This is the current location.
 |}): Promise<{|
   saveAsLocation: ?SaveAsLocation, // This is the newly chosen location (or null if cancelled).
+  saveAsOptions: ?SaveAsOptions,
 |}> => {
-  const defaultPath = fileMetadata ? fileMetadata.fileIdentifier : '';
+  const options = await new Promise(resolve => {
+    setDialog(() => (
+      <SaveAsOptionsDialog
+        onCancel={() => {
+          closeDialog();
+          resolve(null);
+        }}
+        nameSuggestion={
+          fileMetadata ? `${project.getName()} - Copy` : project.getName()
+        }
+        mainActionLabel={<Trans>Continue</Trans>}
+        shouldAskForGameLinkRemoval={!!fileMetadata && !!fileMetadata.gameId}
+        onSave={options => {
+          closeDialog();
+          resolve(options);
+        }}
+      />
+    ));
+  });
+
+  if (!options) return { saveAsLocation: null, saveAsOptions: null }; // Save was cancelled.
+
+  let defaultPath = fileMetadata ? fileMetadata.fileIdentifier : '';
+  if (path && defaultPath) {
+    defaultPath = path.join(path.dirname(defaultPath), `${options.name}.json`);
```

Comment:
any risk the `options.name` results in a filename that is not compatible with path? like with '/' in it?

Reply:
Good remark, I'll add a check on the project name